protocol/messenger_share_urls_test.go

```diff
@@ -369,20 +369,20 @@ func (s *MessengerShareUrlsSuite) TestShareUserURLWithData() {
 	url, err := s.m.ShareUserURLWithData(contact.ID)
 	s.Require().NoError(err)
 
-	userData, signature, err := s.m.prepareEncodedUserData(contact)
+	userData, chatKey, err := s.m.prepareEncodedUserData(contact)
 	s.Require().NoError(err)
 
-	expectedURL := fmt.Sprintf("%s/u/%s#%s", baseShareURL, userData, signature)
+	expectedURL := fmt.Sprintf("%s/u/%s#%s", baseShareURL, userData, chatKey)
 	s.Require().Equal(expectedURL, url)
```

Comment:
I think the asserts could improved or at leats maybe we could add a new assert.

The one you do right now basically just checks that `ShareUserURLWithData` returns the URL with the right order, because both use `prepareEncodedUserData`.

It would be good to test that the encoding methods do the right thing by having a "hardcoded" result.

eg:
```go
s.Require().Equal("https://status.app/cc/dfiohf32hf92hfhf30jksapofpok", url)
```

Reply:
yeah, that was a first plan :) but when we create community or contact we should get different ids. At the end, `Shared***Data` methods trying to get community or contact from db. The one thing which I can improve - check that generated data is equal to hardcoded string (except channel data cos it embeds `uuid`)